src/wazuh_modules/syscollector/sysInfoLinux.cpp

```diff
@@ -44,6 +55,176 @@ static bool getSystemInfo(const std::string& fileName, const std::string& separa
+    std::fstream file{fileName, std::ios_base::in};
+    if (file.is_open())
+    {
+        while(file.good())
+        {
+            std::string line;
+            std::vector<std::string> data;
+            do
+            {
+                std::getline(file, line);
+                if(line.front() == ' ')//additional info
+                {
+                    data.back() = data.back() + line + "\n";
+                }
+                else
+                {
+                    data.push_back(line + "\n");
+                }
+            }
+            while(!line.empty());//end of package item info
+            const auto packageInfo{ parsePackage(data) };
+            if (!packageInfo[0].empty())
+            {
+                ret.push_back(packageInfo[0]);
+            }
+        }
+    }
+    return ret;
+}
+
+static nlohmann::json parseRpm(const std::string& packageInfo)
+{
+    nlohmann::json ret;
+    std::string token;
+    std::istringstream tokenStream{ packageInfo };
+    std::map<std::string, std::string> info;
+    while (std::getline(tokenStream, token))
+    {
+        const auto pos{token.find(":")};
+        if (pos != std::string::npos && (pos + 1) < token.size())
+        {
+            const auto key{Utils::trim(token.substr(0, pos))};
+            const auto value{Utils::trim(token.substr(pos + 1))};
+            info[key] = value;
+        }
+    }
+    auto it{info.find("Name")};
+    if (it != info.end() && it->second != "gpg-pubkey")
+    {
+        std::string version;
+        ret["name"] = it->second;
+        it = info.find("Size");
+        if (it != info.end())
+        {
+            ret["size"] = it->second;
+        }
+        it = info.find("Install Date");
+        if (it != info.end())
+        {
+            ret["install_time"] = it->second;
+        }
+        it = info.find("Group");
+        if (it != info.end())
+        {
+            ret["group"] = it->second;
+        }
+        it = info.find("Epoch");
+        if (it != info.end())
+        {
+            version += it->second + "-";
+        }
+        it = info.find("Release");
+        if (it != info.end())
+        {
+            version += it->second + "-";
+        }
+        it = info.find("Version");
+        if (it != info.end())
+        {
+            version += it->second;
+        }
+        ret["version"] = version;
+    }
+    return ret;
+}
+
+static nlohmann::json getRpmInfo()
+{
+    nlohmann::json ret;
+    auto rawData{ Utils::exec("rpm -qai") };
+    if (!rawData.empty())
+    {
+        std::vector<std::string> packages;
+        auto pos{rawData.rfind("Name")};
+        while(pos != std::string::npos)
+        {
+            const auto package{parseRpm(rawData.substr(pos))};
+            if (!package[0].empty())
+            {
+                ret.push_back(package[0]);
+            }
+            rawData = rawData.substr(0, pos);
+            pos = rawData.rfind("Name");
+        }
+    }
+    else
```

Comment:
Do we need this else? If not, should we add a comment saying why else is not needed or is not needed at all?

Reply:
will do